java-client/src/integration-test/java/com/cumulocity/sdk/client/inventory/BinariesApiIT.java

```diff
@@ -36,6 +36,22 @@ public void shouldUploadFile() {
         assertThat(uploaded.get("contentType")).isEqualTo(MediaType.APPLICATION_OCTET_STREAM);
     }
 
+    @Test
+    public void shouldUploadFileWithCustomMimeType() {
+        // given
+        ManagedObjectRepresentation container = new ManagedObjectRepresentation();
+        container.setName("java-sdk_sample_binary");
+        container.setType("sag_Binary");
+        container.set("iot.cumulocity.com", "domain");
+
+        InputStream file = getFileInputStream("sampleTestFile.txt");
+        // when
+        ManagedObjectRepresentation uploaded = binariesApi.uploadFile(container, file, MediaType.TEXT_PLAIN_TYPE);
```

Comment:
Do we have tests for other MediaTypes? 
Why the TEXT_PLAIN_TYPE is consider as a custom value? This is value from list of `javax.ws.rs.core.MediaType` standard types. So when I see custom I think about something like `text/cumulo-type`

Reply:
"custom" here means that it's different than the default one used by sdk (application/octet-stream) - we have test for calling upload method without specifying any media type.